README.md

```diff
@@ -1,19 +1,218 @@
+
+### Policy Enable
+
+Each individual policy configuration file (see below) also contains the exact
+same `optConfig` configuration object. This allows granularity to enable
+policies on individual repositories. A policy will not take action unless both
+it is enabled and Allstar is enabled as a whole.
+
+### Definition
+
+https://pkg.go.dev/github.com/ossf/allstar@v0.0.0-20210728182754-005854d69ba7/pkg/config#OrgOptConfig
+https://pkg.go.dev/github.com/ossf/allstar@v0.0.0-20210728182754-005854d69ba7/pkg/config#RepoOptConfig
+
+## **Actions**
+
+Each policy can be configured with an action that Allstar will take when it
+detects a repository to be out of compliance.
+
+- `log`: This is the default action, and actually takes place for all
+  actions. All policy run results and details are logged. Logs are currently
+  only visible to the app operator, plans to expose these are under discussion.
+- `issue`: This action creates a GitHub issue. Only one issue is created per
+  policy, and the text describes the details of the policy violation. If the
+  issue is already open, it is pinged with a comment every 24 hours (not
+  currently user configurable). Once the violation is addressed, the issue
+  should be closed by Allstar within 5-10 minutes.
+- `fix`: This action is policy specific. The policy will make the changes to the
+  GitHub settings to correct the policy violation. Not all policies will be able
+  to support this (see below).
+
+Proposed, not yet implemented, actions. Further discussion and definition needed.
+
+- `block`: Allstar can set a [GitHub Status
+  Check](https://docs.github.com/en/github/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks)
+  and block any PR in the repository from being merged.
+- `email`: Allstar could send an email somehow.
+- `rpc`: Allstar could send an rpc to some organization-specific system.
+
+## **Policies**
+
+Similar to the Allstar app enable configuration, all policies are enabled and
+configured with a yaml file in either the organization's `.allstar` repository,
+or the repository's `.allstar` directory. As with the app, policies are opt-in
+by default, also the default `log` action won't produce visible results. A
+simple way to enable all policies is to create a yaml file for each policy with
+the contents:
+
+```
+optConfig:
+  optOutStrategy: true
+action: issue
+```
+
+
+### Branch Protection
+
+This policy's config file is named `branch_protection.yaml`, and the [config
+definitions are
+here](https://pkg.go.dev/github.com/ossf/allstar@v0.0.0-20210728182754-005854d69ba7/pkg/policies/branch#OrgConfig).
+
+The branch protection policy checks that GitHub's [branch protection
+settings](https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)
+are setup correctly according to the specified configuration. The `fix` action
```

Comment:
"this fix action is not yet implemented" can be just added in end, rather than repeating in every policy.

Reply:
Instead add a line on how to manually fix these checks, that part is important as these users wont know. (in each of the checks).

Reply:
done